docs/4.4/api-reference.md

```diff
@@ -11,57 +11,287 @@ Teleport is currently working on documenting our API.
-    logins: ['access-plugin']
-version: v3
+user with appropriate permissions. You should only give the api user permissions for what it actually needs. 
+
+To quickly get started with the api, you can use this api-admin user, but in real usage make sure to have stringent permissions in place.
+
+```yaml
+# Copy and Paste the below and run on the Teleport Auth server.
+$ cat > api-admin.yaml <<EOF
+{!examples/go-client/api-admin.yaml!}
 EOF
-# ...
-$ tctl create rscs.yaml
-# ...
-$ tctl auth sign --format=tls --user=access-plugin --out=auth
-# ...
+
+$ tctl create -f api-admin.yaml
+$ mkdir -p certs
+$ tctl auth sign --format=tls --user=api-admin --out=certs/api-admin 
 ```
 
-The above sequence should result in three PEM encoded files being generated:
-`auth.crt`, `auth.key`, and `auth.cas` (certificate, private key, and CA certs respectively).
+This should result in three PEM encoded files being generated in the `/certs` directory: `api-admin.crt`, `api-admin.key`, and `api-admin.cas` (certificate, private key, and CA certs respectively).
 
-Note: by default, tctl auth sign produces certificates with a relatively short lifetime.
-For production deployments, the --ttl flag can be used to ensure a more practical
-certificate lifetime.
+!!! Note
+    By default, `tctl auth sign` produces certificates with a relatively short lifetime.
+    For production deployments, the `--ttl` flag can be used to ensure a more practical
+    certificate lifetime.
 
-# gRPC APIs
+## Setup Go Client
+
+Put `client.go` (below) in the same folder as `certs`. Then you can create a `main.go` file and use the client from `connectClient` to run the following examples, or experiment on your own. Your folder structure should look like this:
+
+```
+.
++-- api-admin.yaml
++-- certs
+|   +-- api-admin.cas
+|   +-- api-admin.crt
+|   +-- api-admin.key
++-- client.go
++-- main.go
+```
+
+**client.go**
+```go
+{!examples/go-client/client.go!}
+```
+
+## Roles
+
+[Roles](enterprise/ssh-rbac.md#roles) are used to define what resources and actions a user is allowed/denied access to. It's best to make your role's permissions as stringent as possible.
+
+**Create a new role**
+
+```go
+// create a new auditor role which has very limited permissions
+role, err := services.NewRole("auditor", services.RoleSpecV3{
+  Options: services.RoleOptions{
+    MaxSessionTTL: services.Duration(time.Hour),
+  },
+  Allow: services.RoleConditions{
+    Logins: []string{"auditor"},
+    Rules: []services.Rule{
+      // - resources: ['session']
+      //   verbs: ['list', 'read']
+      services.NewRule(services.KindSession, services.RO()),
+    },
+  },
+  Deny: services.RoleConditions{
+    // node_labels: '*': '*'
+    NodeLabels: services.Labels{"*": []string{"*"}},
+  },
+})
+if err != nil {
+  return err
+}
+
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Retrieve role**
+
+```go
+role, err := client.GetRole("auditor")
+if err != nil {
+  return err
+}
+```
```

Comment:
I would always structure API Example in this order, as it should be the simplest. Read, Create, Update, Delete